hnn_core/network.py

```diff
@@ -124,6 +124,37 @@ def _compare_lists(s, t):
     return not t
 
 
+def _update_dict_keys(input_dict, old_key, new_key):
+    """Recursively update keys and values in a dictionary.
+
+    Parameters
+    ----------
+    input_dict : dict
+        The dictionary to update.
+    old_key : str
+        The key or value to replace.
+    new_key : str
+        The new key or value to replace with.
+
+    Returns
+    -------
+    updated_dict : dict
+        The updated dictionary with keys and values replaced.
+    """
+    updated_dict = dict()
+    for key, value in input_dict.items():
```

Comment:
note after python 3.7 or 3.8 (don't remember the exact version) this will preserve order